receiver/flinkmetricsreceiver/scraper.go

```diff
@@ -0,0 +1,106 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flinkmetricsreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/flinkmetricsreceiver"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/flinkmetricsreceiver/internal/metadata"
+)
+
+var errClientNotInit = errors.New("client not initialized")
+
+type flinkmetricsScraper struct {
+	client   client
+	cfg      *Config
+	settings component.TelemetrySettings
+	mb       *metadata.MetricsBuilder
+}
+
+func newflinkScraper(config *Config, settings component.TelemetrySettings) *flinkmetricsScraper {
+	return &flinkmetricsScraper{
+		settings: settings,
+		cfg:      config,
+		mb:       metadata.NewMetricsBuilder(config.Metrics),
+	}
+}
+
+func (s *flinkmetricsScraper) start(_ context.Context, host component.Host) error {
+	httpClient, err := newClient(s.cfg, host, s.settings, s.settings.Logger)
+	if err != nil {
+		return fmt.Errorf("failed to start: %w", err)
+	}
+	s.client = httpClient
+	return nil
+}
+
+func (s *flinkmetricsScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
+	// Validate we don't attempt to scrape without initializing the client
+	if s.client == nil {
+		return pmetric.NewMetrics(), errClientNotInit
+	}
+
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	jobmanagerMetrics, err := s.client.GetJobmanagerMetrics(ctx)
+	if err != nil {
+		s.settings.Logger.Error("Failed to fetch jobmanager metrics",
+			zap.String("endpoint", s.cfg.Endpoint),
+			zap.Error(err),
+		)
+		return pmetric.NewMetrics(), err
+	}
+
+	taskmanagersMetrics, err := s.client.GetTaskmanagersMetrics(ctx)
+	if err != nil {
+		s.settings.Logger.Error("Failed to fetch taskmanager metrics",
+			zap.String("endpoint", s.cfg.Endpoint),
+			zap.Error(err),
+		)
+		return pmetric.NewMetrics(), err
+	}
```

Comment:
If a jobmanager fails, an error should be returned since the others sources depend on the jobmanager. If taskmanagers, jobs, or subtasks calls fail, then a partial error will be added and partial metrics will be returned.